js/msp/MSPHelper.js

```diff
@@ -0,0 +1,1811 @@
+                var offset;
+                for (offset = 0; offset < 4; offset++) {
+                    identifier += String.fromCharCode(data.getUint8(offset));
+                }
+                CONFIG.boardIdentifier = identifier;
+                CONFIG.boardVersion = data.getUint16(offset, 1);
+                offset+=2;
+                break;
+    
+            case MSPCodes.MSP_NAME:
+                var offset = 0;
+                var name = '';
+                while (offset<data.byteLength) {
+                    name += String.fromCharCode(data.getUint8(offset++));;
+                }
+                CONFIG.name = name;
+                break;
+    
+            case MSPCodes.MSP_SET_CHANNEL_FORWARDING:
+                console.log('Channel forwarding saved');
+                break;
+    
+            case MSPCodes.MSP_CF_SERIAL_CONFIG:
+                var supportedBaudRates = [ // 0 based index.
+                                           'AUTO',
+                                           '9600',
+                                           '19200',
+                                           '38400',
+                                           '57600',
+                                           '115200',
+                                           '230400',
+                                           '250000',
+                                       ];
+                if (semver.lt(CONFIG.apiVersion, "1.6.0")) {
+                    SERIAL_CONFIG.ports = [];
+                    var offset = 0;
+                    var serialPortCount = (data.byteLength - (4 * 4)) / 2;
+                    for (var i = 0; i < serialPortCount; i++) {
+                        var serialPort = {
+                            identifier: data.getUint8(offset++, 1),
+                            scenario: data.getUint8(offset++, 1)
+                        }
+                        SERIAL_CONFIG.ports.push(serialPort); 
+                    }
+                    SERIAL_CONFIG.mspBaudRate = data.getUint32(offset, 1);
+                    offset+= 4;
+                    SERIAL_CONFIG.cliBaudRate = data.getUint32(offset, 1);
+                    offset+= 4;
+                    SERIAL_CONFIG.gpsBaudRate = data.getUint32(offset, 1);
+                    offset+= 4;
+                    SERIAL_CONFIG.gpsPassthroughBaudRate = data.getUint32(offset, 1);
+                    offset+= 4;
+                } else {
+                    SERIAL_CONFIG.ports = [];
+                    var offset = 0;
+                    var bytesPerPort = 1 + 2 + (1 * 4);
+                    var serialPortCount = data.byteLength / bytesPerPort;
+                    
+                    for (var i = 0; i < serialPortCount; i++) {
+                        var serialPort = {
+                            identifier: data.getUint8(offset, 1),
+                            functions: MspHelper.serialPortFunctionMaskToFunctions(data.getUint16(offset + 1, 1)),
```

Comment:
This will break now, since `MspHelper` is not the name of the global var anymore. In any case, referencing a global var in here is not robust.
Rather, do `var self = this'` at the beginning of the method, then go `self.serialPortFunctionMaskToFunctions`.